sanskrit_parser/lexical_analyzer/sanskrit_lexical_analyzer.py

```diff
@@ -53,20 +58,21 @@ def hasNode(self, t):
                boolean
         '''
         return t in self.G
-    def appendToNode(self, t, rdag):
-        """ append rdag to self, adding edges from a given node to rdag's roots 
+
+    def appendToNode(self, t, nodes):
+        """ Create edges from t to nodes
 
             Params:
                 t (SanskritObject)      : Node to append to
-             rdag (SanskritLexicalGraph): Graph to append to node
+                nodes (iterator(nodes)) : Nodes to append to t
         """
         # t is in our graph
         assert t in self.G
-        self.G = nx.compose(self.G, rdag.G)
-        for r in rdag.roots:
+        for r in nodes:
             self.G.add_edge(t, r)
```

Comment:
The above is the main change to avoid the composition and just directly add edges from r to the root nodes of rdag.

Reply:
The addNode signature has been changed, but __init__ uses the old signature.

Reply:
```
        self.splits = SanskritLexicalGraph()
        roots = self._possible_splits(s)
        if tag and len(roots) > 0:
            self.tagLexicalGraph(self.splits)
        if len(roots) == 0:
            return None
        else:
            self.splits.addRoots(roots)
            return self.splits
```
self.splits is empty when tagLexicalGraph is called. This will break the morphological analyzer

Reply:
Looks like `__init__` is never called with any arguments. If there is no objection, I will just remove the arguments and the code that calls addNode

I am not sure I understand the comment regarding the morphological analyzer. `self._possible_splits` adds all the requisite nodes into the graph. `self.splits` is not reset at any time unless getSandhiSplits is called again with a different argument, at which time it will be updated with the graph for the splits of the new string. I have confirmed that the morphological analyzer still works with these changes.

Perhaps the change hides the fact that `_possible_splits` updates `self.splits`. We could rename the function if that is the confusion, or add a comment at that line.

Reply:
You're right, I hadn't noticed that self.splits was being updated by _possible_splits when I wrote that comment. tagLexicalGraph does not need edges updated to work right. 
Should be ok then. 
>Looks like __init__ is never called with any arguments. If there is no objection, I will just remove the arguments and the code that calls addNode
Yes please.